### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ychaiun**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const arr = [];
```
Comment (polosanya) [inline src/transformStateWithClones.js:11]: Use proper names. Specify what exactly this array is

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -7,7 +7,31 @@
...
+
+  for (const action of actions) {
+    switch (action.type) {
```
Comment (polosanya) [inline src/transformStateWithClones.js:14]: You can destructure the action to make your code a bit cleaner

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -7,7 +7,31 @@
...
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in newState) {
+          delete newState[key];
+        }
```
Comment (polosanya) [inline src/transformStateWithClones.js:28]: Or you can just reassign the newState as empty object
`newState = {}`

**Review by polosanya — CHANGES_REQUESTED** [review]
Let's improve your code a bit)

**Review by agniya-i — APPROVED** [review]

**Pull request closed without merging**